### ISSUE TITLE
subscripts of pi'a and sa'i

### ISSUE THREAD

**Issue by phma** [lojban/cll#455]
Section 18.15 states "Matrices of more than two dimensions can be built up using either pi'a or sa'i with an appropriate subscript numbering the dimension." Which subscript is pi'a and which is sa'i?

**Comment by lagleki** [comment]
On Fri, Mar 19, 2021, 13:40 Pierre Abbat ***@***.***> wrote:

> Section 18.15 states "Matrices of more than two dimensions can be built up
> using either pi'a or sa'i with an appropriate subscript numbering the
> dimension." Which subscript is pi'a and which is sa'i?
>


It doesn't matter in case you describe dimension > 2.

Or are you asking whether sa'i= pi'axire?

**Comment by phma** [comment]
Yes. It's not clear whether subscripts start at 0 or 1 or whether pi'a or sa'i has the smaller subscript. So it could be:

- pi'a=pi'axino, sa'i=pi'axipa
- sa'i=pi'axino, pi'a=pi'axipa
- pi'a=pi'axipa, sa'i=pi'axire
- sa'i=pi'axipa, pi'a=pi'axire.

**Comment by lagleki** [comment]
"numbering the dimension" implies vectors being of dimension 1 should have
{xipa}.
However, it might be beneficial to have {xi mlatu bu} form as an
alternative. See
https://pytorch.org/docs/stable/named_tensor.html

**Comment by lagleki** [comment]
@phma what do you think of this fix?
```
Matrices of more than two dimensions can be built up using either
      <valsi>pi'a</valsi>
      or
      <valsi>sa'i</valsi>
      with an appropriate subscript labeling the dimension. When subscripted, there is no difference between
      <valsi>pi'a</valsi>
      and
      <valsi>sa'i</valsi>. Labels can be any anything that <valsi>xi</valsi> supports, e.g. <valsi>pa</valsi> or <valsi>mlatu bu</valsi>.
```

**Comment by phma** [comment]
Still doesn't answer the question. Is {pi'a} equivalent to {pi'axino}, {pi'axipa}, or {pi'axire}? Likewise for {sa'i}?

**Comment by lagleki** [comment]
I dont think they are necessarily equivalent. Dimensions can be arbitrary labels.
